modules/apps/fragment/fragment-test/src/testIntegration/java/com/liferay/fragment/internal/upgrade/v3_0_0/test/FragmentEntryLinkUpgradeProcessTest.java

```diff
@@ -0,0 +1,159 @@
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.Layout;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+import com.liferay.segments.service.SegmentsExperienceLocalService;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Alberto Sousa
+ */
+@RunWith(Arquillian.class)
+public class FragmentEntryLinkUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+
+		_layout = LayoutTestUtil.addTypeContentLayout(_group);
+
+		_segmentsExperienceId =
+			_segmentsExperienceLocalService.fetchDefaultSegmentsExperienceId(
+				_layout.getPlid());
+	}
+
+	@Test
+	public void testUpgrade() throws Exception {
+		String oldCollectionFieldId =
+			"ObjectRelationship#C_ParentObjectDefinition#" +
+				"oneToManyObjectRelationship_textObjectField.getData()";
+		String expectedCollectionFieldId =
+			"ObjectRelationship_C_ParentObjectDefinition_" +
+				"oneToManyObjectRelationship_textObjectField.getData()";
+
+		JSONObject collectionFieldIdJSONObject = JSONUtil.put(
+			"collectionFieldId", oldCollectionFieldId);
+
+		JSONObject elementTextJSONObject = JSONUtil.put(
+			"element-text", collectionFieldIdJSONObject);
+
+		JSONObject editableValuesJSONObject = JSONUtil.put(
+			FragmentEntryProcessorConstants.
+				KEY_EDITABLE_FRAGMENT_ENTRY_PROCESSOR,
+			elementTextJSONObject);
+
+		long fragmentEntryLinkId =
+			ContentLayoutTestUtil.addFragmentEntryLinkToLayout(
+				editableValuesJSONObject.toString(), _layout.fetchDraftLayout(),
+				_segmentsExperienceId
+			).getFragmentEntryLinkId();
+
+		_runUpgrade();
+
+		FragmentEntryLink upgradedFragmentLink =
+			_fragmentEntryLinkLocalService.getFragmentEntryLink(
+				fragmentEntryLinkId);
+
+		String newCollectionFieldId = _getCollectionFieldId(
+			upgradedFragmentLink.getEditableValues());
+
+		Assert.assertEquals(expectedCollectionFieldId, newCollectionFieldId);
+
+		_fragmentEntryLinkLocalService.deleteFragmentEntryLink(
+			fragmentEntryLinkId);
+	}
+
+	private String _getCollectionFieldId(String editableValues)
+		throws Exception {
```

Comment:
Refactor this method to:
1. Add parameters to receive the expected value.
2. Get the FragmentEntryLink.
3. Reuse the existing code to get `collectionFieldId`.
4. Assert the `collectionFieldId` value

Reply:
Rename to `private void _assertFragmentEntryLinkCollectionFieldId()`

Reply:
we will use this method before and after the update